extensions/mssql/src/util/dispose.ts

```diff
@@ -0,0 +1,42 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+
+export function disposeAll(disposables: vscode.Disposable[]) {
+	while (disposables.length) {
+		const item = disposables.pop();
+		if (item) {
+			item.dispose();
+		}
+	}
+}
+
+export abstract class Disposable {
```

Comment:
why do we need to add this?

Reply:
I ported over the Markdown code and it uses it since there's a listener set up that needs to be disposed. So, we make the BookContributionProvider a disposable, and register it so on deactive it disposes as expected. Basically copied a useful utility "Disposable" implementation from Markdown.